src/wp-includes/default-filters.php

```diff
@@ -192,6 +192,7 @@
 add_filter( 'the_title', 'convert_chars' );
 add_filter( 'the_title', 'trim' );
 
+add_filter( 'the_content', 'apply_block_hooks_to_post_content', 8 ); // BEFORE do_blocks().
```

Comment:
Instead of introducing a new function, we could use the existing `apply_block_hooks_to_content`, and set its `$context` argument to default to `get_post()` 🤔